framework/structures.md

```diff
@@ -122,10 +131,12 @@ other AWS services.
 
 #### Implementations
 
-- [Java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/model/EncryptionMaterials.java)
-- [Python](https://github.com/aws/aws-encryption-sdk-python/blob/master/src/aws_encryption_sdk/materials_managers/__init__.py)
-- [C](https://github.com/aws/aws-encryption-sdk-c/blob/master/include/aws/cryptosdk/materials.h)
-- [Javascript](https://github.com/awslabs/aws-encryption-sdk-javascript/blob/master/modules/material-management/src/cryptographic_material.ts)
+| Language | Confirmed Compatible with Spec Version | Minimum Version Confirmed | Implementation |
+|----------|----------------------------------------|---------------------------|----------------|
+| C | 0.1.0-preview | 0.1.0 | [materials.h](https://github.com/aws/aws-encryption-sdk-c/blob/master/include/aws/cryptosdk/materials.h) |
+| Javascript | 0.1.0-preview | 0.1.0 | [cryptographic_material.ts](https://github.com/awslabs/aws-encryption-sdk-javascript/blob/master/modules/material-management/src/cryptographic_material.ts)|
+| Python | 0.1.0-preview | n/a | [materials_managers](https://github.com/aws/aws-encryption-sdk-python/blob/master/src/aws_encryption_sdk/materials_managers/__init__.py) |
+| Java | 0.1.0-preview | n/a | [EncryptionMaterials.java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/model/EncryptionMaterials.java) |
```

Comment:
LOVE having this codified in a table, but I think this will quickly get much more complicated as we have multiple versions in flight. Would it make more sense to have a separate table for each implementation, so that we can have a row for each spec version and the corresponding minimum implementation version?

I realize this is a multi-dimensional matrix and I'm not sure what the optimal rendering is off-hand :)

Reply:
My intention is that the spec feature doc always serves as the source of truth for that is tru *right now*. I like the idea of a historical "implementation X complied with feature version X from implementation versions from version M to version N" for every implementation and feature version, but that feels to me like something that would be better represented in a machine-readable form. Say, a separate JSON file.